larreco/TrackFinder/PMAlgTrackMaker_module_continued.h

```diff
@@ -0,0 +1,1039 @@
+void PMAlgTrackMaker::produce(art::Event& evt)
+{
+
+	fDetProp = lar::providerFrom<detinfo::DetectorPropertiesService>();
+
+	reset(evt); // set default values, clear containers at the beginning of each event
```

Comment:
I think it would be good to consider removing this from the module state, and to have the objects in question be function-scope -- so that they are destroyed upon exit from the function. This guarantees the data do not linger in memory longer than needed, and provides exception safety for free.